src/main/java/io/lettuce/core/cluster/RedisClusterClient.java

```diff
@@ -936,6 +938,36 @@ protected Partitions loadPartitions() {
         return get(loadPartitionsAsync(), Function.identity());
     }
 
+    public void reauthenticateAllConnections(RedisCredentials creds) {
+        forEachClusterConnection(input -> {
+            input.reauthenticate(creds);
+        });
+
+        forEachClusterPubSubConnection(input -> {
+            input.reauthenticate(creds);
+        });
+    }
+    
+    public void reauthInConnections() {
+        RedisCredentialsProvider credentialsProvider = getFirstUri().getCredentialsProvider();
+        if (credentialsProvider instanceof RedisCredentialsProvider.ImmediateRedisCredentialsProvider) {
+            RedisCredentials redisCredentials =
+                ((RedisCredentialsProvider.ImmediateRedisCredentialsProvider)credentialsProvider).resolveCredentialsNow();
+            if (!redisCredentials.hasPassword()) {
+                return;
```

Comment:
should this be an exception? Why silently fail?

Reply:
I don't see it as a failure

Reply:
If the user has setup a client with periodic reatuhentication, but without the ability to authenticate, it's a logical inconsistency. You can decide that you silently ignore it, but that's opening a way for bugs, since the user might've intended for the authentication to happen. You don't ignore inconsistencies, you highlight them so that the user can corect their config.